TypeScript compiler error in master

We're now seeing the following failure in master:

https://travis-ci.com/github/pulumi/pulumi/jobs/328314403#L1400-L1418

```
1401tests/unwrap.spec.ts:54:37 - error TS2769: No overload matches this call.
1402  Overload 1 of 2, '(value: Resource): boolean', gave the following error.
1403    Argument of type 'Resource | TestResource' is not assignable to parameter of type 'Resource'.
1404      Type 'TestResource' is missing the following properties from type 'Resource': __parentResource, __childResources, urn, __protect, and 2 more.
1405  Overload 2 of 2, '(value: TestResource): boolean', gave the following error.
1406    Argument of type 'Resource | TestResource' is not assignable to parameter of type 'TestResource'.
1407      Property 'name' is missing in type 'Resource' but required in type 'TestResource'.
1408
140954             if (!asyncResources.has(res)) {
1410                                       ~~~
1411
1412  tests/unwrap.spec.ts:66:17
1413    66     constructor(public name: string) {
1414                       ~~~~~~~~~~~~~~~~~~~
1415    'name' is declared here.
1416
1417
1418Found 1 error.
```

It's not clear to me why this just recently started erroring. I thought it might be an update to the compiler, but nothing obvious is jumping out comparing [v3.7.3 and v3.7.5](https://github.com/microsoft/TypeScript/compare/v3.7.3...v3.7.5).

--- a/sdk/nodejs/tests/unwrap.spec.ts
+++ b/sdk/nodejs/tests/unwrap.spec.ts
@@ -51,7 +51,7 @@ function testResources(val: any, expected: any, resources: TestResource[], allRe
         assert.deepStrictEqual(asyncResources, new Set(allResources));
 
         for (const res of syncResources) {
-            if (!asyncResources.has(res)) {
+            if (!asyncResources.has(<TestResource>res)) {
                 assert.fail(`async resources did not contain: ${(<TestResource><any>res).name}`)
             }
         }

